comp/core/tagger/tagger.go

```diff
@@ -324,25 +327,89 @@ func (t *TaggerClient) ResetCaptureTagger() {
 // NOTE(remy): it is not needed to sort/dedup the tags anymore since after the
 // enrichment, the metric and its tags is sent to the context key generator, which
 // is taking care of deduping the tags while generating the context key.
-func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, udsOrigin string, clientOrigin string, cardinalityName string) {
-	cardinality := taggerCardinality(cardinalityName)
+func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, originInfo taggertypes.OriginInfo) {
+	cardinality := taggerCardinality(originInfo.Cardinality)
+
+	switch originInfo.ProductOrigin {
```

Comment:
Would it be possible to split this into two functions to avoid shipping product type with each metric sample? AIUI the codepaths for different products are different, we can make this distinction statically in the code, instead of a run time check.

Reply:
Seen together, all good for now.